### ISSUE TITLE
Can we use boolean when searching by text?

### ISSUE THREAD

**Issue by bigheadmo** [devinit/D-Portal#623]
Hello, I am trying to search all the aid programs for example that have the terms 'hearing aids' or 'learning difficulties in them. But first, I am not able to use boolean so need to try each term separately. Maybe there is a way already, but I tried using * or '', combing with AND, OR, but none of them worked. And secondly, when I enter hearing aids, all the entries that have the word 'hearing' would show up.  I can't find an efficient way to limit the search result. Many entries also have missing data of the policy markers so that makes using the policy marker to filter the result quite useless. Please let me know if I am doing anything wrong, or what is the most efficient way to do this. Thank you.

**Comment by xriss** [comment]
Yes you can but even better, since I have not looked at this for a while and due to postgres going up a few versions, it is now easy to switch to slightly friendlier version. So I did.

Which means what you tried before should now work

Note that you need to use " rather than ' so try the following

"hearing aids" or "learning difficulties"

http://d-portal.org/ctrack.html#view=search&text_search=%22hearing%20aids%22%20or%20%22learning%20difficulties%22

Will get 101 results, hope that helps.

For a more technical description search is now done using a simple "websearch_to_tsquery" which you can find documented on this page http://www.postgresql.cn/docs/11/textsearch-controls.html

**Comment by bigheadmo** [comment]
> Yes you can but even better, since I have not looked at this for a while and due to postgres going up a few versions, it is now easy to switch to slightly friendlier version. So I did.
> 
> Which means what you tried before should now work
> 
> Note that you need to use " rather than ' so try the following
> 
> "hearing aids" or "learning difficulties"
> 
> http://d-portal.org/ctrack.html#view=search&text_search=%22hearing%20aids%22%20or%20%22learning%20difficulties%22
> 
> Will get 101 results, hope that helps.
> 
> For a more technical description search is now done using a simple "websearch_to_tsquery" which you can find documented on this page http://www.postgresql.cn/docs/11/textsearch-controls.html

Thank you so much for such quick turnaround. but when I look closer at each entry. For example, this one: http://d-portal.org/ctrack.html?text_search=%22hearing%20aids%22%20or%20%22learning%20difficulties%22#view=act&aid=CA-3-D000252001
I am wonderirng why it shows up in the search result when there is no text 

> Yes you can but even better, since I have not looked at this for a while and due to postgres going up a few versions, it is now easy to switch to slightly friendlier version. So I did.
> 
> Which means what you tried before should now work
> 
> Note that you need to use " rather than ' so try the following
> 
> "hearing aids" or "learning difficulties"
> 
> http://d-portal.org/ctrack.html#view=search&text_search=%22hearing%20aids%22%20or%20%22learning%20difficulties%22
> 
> Will get 101 results, hope that helps.
> 
> For a more technical description search is now done using a simple "websearch_to_tsquery" which you can find documented on this page http://www.postgresql.cn/docs/11/textsearch-controls.html

Thank you so much for the quick reply. However, when I look closer at each program, I am wondering why they showed up in the result. for example this one: http://d-portal.org/ctrack.html?text_search=%22hearing%20aids%22%20or%20%22learning%20difficulties%22#view=act&aid=CA-3-D000252001

There are no "hearing aids" nor "learning difficulties" mentioned on the whole page. Any ideas? Thank you very much.

**Comment by notshi** [comment]
Thanks for raising this issue.

The search term, "hearing aids" is actually in the data description but we do not currently display it on the page you're looking at yet.

You can see the raw xml here - http://d-portal.org/q.xml?aid=CA-3-D000252001

![Screenshot-20210703022527-1321x138](https://user-images.githubusercontent.com/1515961/124339385-14813c00-dba6-11eb-9c8f-da8c2a401691.png)

**Comment by bigheadmo** [comment]
> Thanks for raising this issue.
> 
> The search term, "hearing aids" is actually in the data description but we do not currently display it on the page you're looking at yet.
> 
> You can see the raw xml here - http://d-portal.org/q.xml?aid=CA-3-D000252001
> 
> ![Screenshot-20210703022527-1321x138](https://user-images.githubusercontent.com/1515961/124339385-14813c00-dba6-11eb-9c8f-da8c2a401691.png)

This is great. So just want to confirm, if I search "hearing aids", "learning difficulties", "web accessibility" and etc, the results will give me the programs that have those full terms in the description. and it will not give me results for example contains "hearing " in one place, and "aid" in another on the same page, correct? not all the programs that contain the word "hearing" will show up, correct? thank you! (I am on the clock here to crunch the numbers, so really appreciate your help)

**Comment by notshi** [comment]
Yes, we believe so. We do recommend double checking for false positives but yes, the search should only give you results with those word combinations.
